style-guide: document not rendered syntax (#12327)
tldr-pages/tldr: contributing-guides/style-guide.md

 `ls`
```

## Not rendered syntax

> Any syntax not documented in this guide or the client specification will **not** be rendered in most of our clients. See common cases below.

Reviewer: This is unclear. What do you mean "any syntax"? Technically the client spec requires CommonMark support, even if we don't use bold/italics etc by convention.....

Reviewer 2: > This is unclear. What do you mean "any syntax"? Technically the client spec requires CommonMark support, even if we don't use bold/italics etc by convention.....

Maybe, we could reword this description to include CommonMark too.

> Any syntax not documented on this guide or in the client specification will **not** be rendered in our clients. See common cases below.
> 
> ### Links and images
> 
> - Do not use Markdown links (\[link text\](link url)) or images in the pages, they are not supported by our clients.

Reviewer 3: ```suggestion
Any syntax not documented on this guide or in the client specification will **not** be rendered in most of our clients. See common cases below.

### Links and images

- Do not use Markdown links (\[link text\](link url)) or images in the pages, they are not part of the tldr client specification and thus not supported by most of our clients.
```

I would refrain from making absolute statements, as we ultimately do not have control over the behaviour of our clients (apart from the things required in the client specification).

Also, the specification states that our pages are written in CommonMark (https://commonmark.org/help/), which explicitly includes the mentioned Link syntax. I would be surprised if none of the clients would render such links as described by CommonMark.

> Any syntax not documented on this guide or in the client specification is not guaranteed to be rendered in our clients. See common cases below.
> 
> ### Links and images
> 
> - Do not use Markdown links (\[link text\](link url)) or images in the pages, they are not guaranteed to be supported by our clients.

Reviewer 2: ```suggestion
Any syntax not documented in this guide or the client specification will **not** be rendered in most of our clients. See common cases below.

### Links and images

- Do not use Markdown links (\[link text\](link url)) or images in the pages, they are not part of the tldr client specification and thus not supported by most of our clients.
```

@gutjuri's suggestions seems to be marked as outdated post merging with main, readded it here along with some minor changes.

> - If you want to use a link in a page, write it normally, either outside or inside parenthesis.

Reviewer 3: ```suggestion
- If you want to use a link in a page, put it in angled brackets (e.g. <https://example.org>).
```

I understand that we usually do it like this as in the more information links.

> 
> - Do not use Markdown links (\[link text\](link url)) or images in the pages, they are not part of the tldr client specification and thus not supported by most of our clients.
> - If you want to use a link in a page, enclose it within angled brackets (e.g. <https://example.org>).

Reviewer: Right but for the wrong reasons? Also, how about this rewording? Too many `do not`s on this page can give the wrong impression when reading.
```suggestion
Use angled brackets (e.g. <https://example.org>) when including links in pages.
```
I would also suggest that just because something isn't supported by clients isn't a good enough reason on its own to avoid it here.

Do we really need to say that images are not allowed? The linter won't even accept them, and unless I'm wrong we haven't ever had a PR that adds an image to a tldr page....

Reviewer 2: > Do we really need to say that images are not allowed? The linter won't even accept them, and unless I'm wrong we haven't ever had a PR that adds an image to a tldr page....

Afaik we haven't had any PR having images using the Markdown rendering format `![alt text](link)`. And I don't think tldr-lint has a rule for it.
